src/frontend/src/views/Data/components/SamplesView/components/SampleTableModalManager/index.tsx

```diff
@@ -34,16 +35,16 @@ const SampleTableModalManager = ({
   // generate ID lists from sample objects
   useEffect(() => {
     const checkedIds = checkedSamples.map((s) => s.publicId);
-    const failedIds = checkedSamples
-      .filter((s) => s.CZBFailedGenomeRecovery)
-      .map((s) => s.publicId);
-    const badQCIds = checkedSamples
+    const badOrFailedQCIds = checkedSamples
       // for now there should only ever be one qcMetrics entry per sample
-      .filter((s) => s.qcMetrics[0].qc_status === "Bad")
+      .filter(
+        (s) =>
```

Comment:
I noticed you used this pattern 3 times in this PR. I think it is worth making a util function that accepts an item, and returns whether its qc status is bad or failed. What do you think?

Reply:
i think it's small enough to go either way, i can make a function